jax/_src/scipy/special.py

```diff
@@ -2653,3 +2653,216 @@ def log_softmax(x: ArrayLike,
     :func:`softmax`
   """
   return nn_log_softmax(x, axis=axis)
+
+
+
+@custom_derivatives.custom_jvp
+def fresnel(x: ArrayLike) -> tuple[Array, Array]:
+  r"""The Fresnel integrals
+
+  JAX implementation of :obj:`scipy.special.fresnel`.
+
+  The Fresnel integrals are defined as
+    .. math::
+       S(x) &= \int_0^x \sin(\pi t^2 /2) dt \\
+       C(x) &= \int_0^x \cos(\pi t^2 /2) dt.
+
+  Args:
+    x: arraylike, real-valued.
+
+  Returns:
+    Arrays containing the values of the Fresnel integrals.
+
+  Notes:
+     The JAX version only supports real-valued inputs, and
+     is based on the SciPy C++ implementation, see
+     `here
+     <https://github.com/scipy/scipy/blob/1abc9faf43d7b82e0b26589bb705d8d009b18713/scipy/special/special/cephes/fresnl.h>`_.
+  """
+  if jnp.iscomplexobj(x):
+    raise NotImplementedError(
+        'Support for complex-valued inputs is not implemented yet.')
+
+  xxa, = promote_args_inexact("fresnel", x)
+
+  # This part is mostly a direct translation of SciPy's C++ code.
+
+  fresnl_sn = jnp.array([
```

Comment:
Sometimes we've found that series expansions developed for 64-bit arithmetic fail in surprising ways when evaluated in 32-bit. Have you checked the precision of the outputs when `jax_enable_x64=False`?

Reply:
I have added a test and the output suggests that the series expansions work for both single and double precision.